nodebuilder/header/service.go

```diff
@@ -51,6 +52,33 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	// TODO(vgonkivs): remove after https://github.com/celestiaorg/go-header/issues/32 will be
+	// implemented
+	head, err := s.store.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()) < height {
```

Comment:
the problem here is that sync head can (and likely will) be > store head in many cases -- therefore we should check if height == syncHead, if not, check if it exceeds localHead and short circuit.

Reply:
We check here against the local head, not the sync head. I don't get why you think we should do syncHead here

Reply:
@Wondertan - so we should not allow sync head to be used in this component? What differentiates this from CoreAccessor for example?

Reply:
We use syncHead here